### PR TITLE
Fix toString() method for tuples

### LINKED ISSUE #32129: ToString call provides wrong value for tuple types
**Description:**
As per the spec, the string value (direct style) of an array should be in the form of `[X,Y]`. 

Currently, this returns the correct value for an array value. But for a tuple value it gives a different value. 

**Steps to reproduce:**

```ballerina
    string[] a = ["a", "b"];
    io:println(a.toString()); // ["a","b"]

    [string, int] b = ["abc", 123];
    io:println(b.toString()); // abc 123 - should be ["abc", 123]
```

**Affected Versions:**
SL beta 2

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
 As per the spec, the string value (direct style) of an array should be in the form of [X,Y].

Fixes #32129 

## Samples
```ballerina
 string[] a = ["a", "b"];
io:println(a.toString()); // ["a","b"]

[string, int] b = ["abc", 123];
io:println(b.toString()); // ["abc", 123]
```
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by HindujaB** [comment]
.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/matchstmt/MatchStatementStaticPatternsTest.java [hunk 1]
```diff
@@ -122,17 +122,17 @@ public void testStaticMatchOrPatterns2() {
...
-        Assert.assertEquals(results.getString(++i), msg + ": 12 Ballerina");
-        Assert.assertEquals(results.getString(++i), msg + ": 12 Ballerina");
-        Assert.assertEquals(results.getString(++i), msg + ": 15 Ballerina");
-        Assert.assertEquals(results.getString(++i), msg + ": 20 Ballerina");
-        Assert.assertEquals(results.getString(++i), msg + ": 20 Bala");
-        Assert.assertEquals(results.getString(++i), "Default Value is : 20 NothingToMatch");
-        Assert.assertEquals(results.getString(++i), msg + ": 15 Bal 100");
-        Assert.assertEquals(results.getString(++i), "Default Value is : 15 Bal 200 230");
-        Assert.assertEquals(results.getString(++i), msg + ": 15 Bal Ballerina 5678 Test");
+        Assert.assertEquals(results.getString(++i), msg + ": [12,Ballerina]");
```
Comment (riyafa) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/matchstmt/MatchStatementStaticPatternsTest.java:125]: shouldn't this be `[12, "Ballerina"]` based on your changes?

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java [hunk 2]
```diff
@@ -390,20 +392,37 @@ public void unshift(Object[] values) {
 
     @Override
     public String stringValue(BLink parent) {
-        StringJoiner sj = new StringJoiner(" ");
+        StringJoiner sj = new StringJoiner(",");
         for (int i = 0; i < this.size; i++) {
-            sj.add(StringUtils.getStringValue(this.refValues[i], new CycleUtils.Node(this, parent)));
+            Object value = this.refValues[i];
```
Comment (lasinicl) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/TupleValueImpl.java:397]: for a `nil` value `toString()` should be `null`. It should be specifically handled here. 
```ballerina
[int?, string] a = [(), "ABC"];
```
`a.toString()` is `[,"ABC"]` ATM, but should be `[null,"ABC"]`

**Review by lasinicl — APPROVED** [review]

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**